pkg/pluginmanager/manager.go

```diff
@@ -697,24 +697,28 @@ func GetPluginGroup(groupIDAndVersion string, options ...PluginManagerOptions) (
 	return pg, nil
 }
 
-func logPluginInstallationMessage(p *discovery.Discovered, version string, isPluginInCache, isPluginAlreadyInstalled bool) {
+func getPluginInstallationMessage(p *discovery.Discovered, version string, isPluginInCache, isPluginAlreadyInstalled bool) (installingMsg, installedMsg string) {
 	withTarget := ""
 	if p.Target != configtypes.TargetUnknown {
-		withTarget = fmt.Sprintf("with target '%v' ", p.Target)
+		withTarget = fmt.Sprintf("with target '%v'", p.Target)
 	}
 
 	if isPluginInCache {
 		if !isPluginAlreadyInstalled {
-			log.Infof("Installing plugin '%v:%v' %v(from cache)", p.Name, version, withTarget)
+			installingMsg = fmt.Sprintf("Installing plugin '%v:%v' %v (from cache)", p.Name, version, withTarget)
+			installedMsg = fmt.Sprintf("Installed plugin '%v:%v' %v (from cache)", p.Name, version, withTarget)
 		} else {
-			log.Infof("Plugin '%v:%v' %vis already installed. Reinitializing...", p.Name, version, withTarget)
+			installingMsg = fmt.Sprintf("Plugin '%v:%v' %v is already installed. Reinitializing...", p.Name, version, withTarget)
+			installedMsg = fmt.Sprintf("Reinitialized plugin '%v:%v' %v", p.Name, version, withTarget)
 		}
 	} else {
-		log.Infof("Installing plugin '%v:%v' %v", p.Name, version, withTarget)
+		installingMsg = fmt.Sprintf("Installing plugin '%v:%v' %v", p.Name, version, withTarget)
+		installedMsg = fmt.Sprintf("Installed plugin '%v:%v' %v", p.Name, version, withTarget)
 	}
+	return installingMsg, installedMsg
 }
 
-func installOrUpgradePlugin(p *discovery.Discovered, version string, installTestPlugin bool) error {
+func installOrUpgradePlugin(p *discovery.Discovered, version string, installTestPlugin, enableSpinner bool) error {
```

Comment:
I see that all the function calls we have are passing `enableSpinner=true`.

Should we have some generic mechanism at runtime to disable this type of dynamic log message (like spinner)? WDYT @chandrareddyp @marckhouzam @vuil ?

If we need to provide some flexibility here, it would be better to control this using some other means (e.g. env variable) instead of passing this as part of this function argument.

Reply:
Agreed.  But first let's clarify the "if we need to provide some flexibility here".  Do we @chandrareddyp ?

Reply:
I felt it would be better to provide flexibility to enable/disable the spinner. Similarly to how we can control NO_COLOR in runtime (here - https://github.com/vmware-tanzu/tanzu-plugin-runtime/blob/8b705913292937d28e37073ae1d8926da19c2380/component/colorable-tty.go#L31), using an environment variable.

Yes, I agree that using an environment variable would be better!

Reply:
code has updated to use the existing runtime API ([component.IsTTYEnabled()](https://github.com/vmware-tanzu/tanzu-plugin-runtime/blob/8b705913292937d28e37073ae1d8926da19c2380/component/colorable-tty.go#L29))